python/GafferUI/_StyleSheet.py

```diff
@@ -1419,6 +1419,19 @@ def styleColor( key ) :
 		margin-right: $toolOverlayInset;
 	}
 
+	*[gafferClass="GafferSceneUI.InstancerUI._VariationsPlugValueWidget"] QLabel
+	{
+		font-family: $monospaceFontFamily;
+		font-weight: bold;
+		font-size : 12pt;
+		color: rgb( 0, 0, 0 );
```

Comment:
We've moved a bit closer to the SceneInspector styling with the rounded backdrops, but there is a difference, in that the SceneInspector displays information someone is looking for, whereas this displays an important warning that needs to stand out amongst the parameters.  I've certainly backed off some on how much it stands out - I understand why we don't want it to be red or yellow - but I don't really like the idea of having it be as small and thin as the SceneInspector readout?

Reply:
I get what you're saying, it's just that this is the largest presentation of any text anywhere in the Gaffer UI and uses a presentation style we don't use anywhere else. I think there are two parts to it color and size. 

Gaffer uses light foreground text everywhere else, can we at least match that? Luminance is generally used in the current theme to indicate z-depth. Broadly speaking lighter elements are 'on top' of darker ones. So making the text black actually pushes is back, rather than making it stand out more. 

On size, our brains tend to filter out the things that they're not interested in. Size in presentation is generally beneficial for fast discovery, when people need to look up information quickly, or be able to glance from a distance. Eg: Newspaper headlines. It also serves as an organisational division, eg: `h1`-`h6` in HTML which provide logical partitioning, which falls back to discovery. Ultimately size is usually part of an information hierarchy. 

So, are we saying those numbers are the things that should be more important than any other text in the Gaffer UI - in terms of someone being able to find what they need to complete their task with the Instancer?

If the primary interaction with an Instancer in the `NodeEditor` is to purely _inspect_ (in a read-only way) how many variations it makes, then the previous presentation where the total was at the top makes more sense, and the layout of the rest of the UI should reorder to reflect that too.

I'd imagine though that most users would be focusing on the controls themselves, and the variation count is the _result_ of a specific configuration (which is why we ended up with the order of presentation we have now). So the controls are the primary interaction point and focus. I'd suggest that users here probably fall into three categories:

 1. Those who don't really know how to apply those numbers to their render cost, and are dialling in a look. They will still make heavy renders regardless how big the information is as they don't know what the information itself means. 
 2. Those who understand render cost. They don't need a slap-in-the face display as they care anyway and will already be taking it into account (or choosing consciously to ignore it).
 3. Those from `2` who understand render cost, and have come to help those in `1` fix their renders.

`2` and `3` need a clear information architecture. I think we have this in the current structure, given were in a left-to-right, top-to-bottom language domain. Controls precede results from left to right, and the combined result is presented below it all. This matches the standard totalling layout from traditional spreadsheets. Your genius idea to present them with the information in the first place is what really helps here. Having varying sizes in the row may even make it harder to follow, as the visual weighting breaks up the logical ordering in the flow of information.

Regarding `2`, as an example, on one shot you may have a 2 million point cache, and be instancing tiny light weight meshes. In this situation, 2000 variations may be totally legit. In another shot, you may have a hundred point cache, instancing detailed buildings, and having 5 variations may be too many, the knowing the number of variations alone won't help fix anyone's renders. They have to know what the number means in context of their work, and that is the bigger hurdle with this tool. Then number '2000' doesn't mean anything in itself, which is why just making it larger doesn't necessarily help solve the problem.

So where might that leave us?

Proposal:

 - Stick with the paradigm that the controls are main focus for `NodeEditor` interactions, and keep the current layout.
 - Make sure that the variations field doesn't look like an editable widget.

As a picture often helps over words, I put together a rough implementation here

![image](https://user-images.githubusercontent.com/896779/105049916-19718980-5a65-11eb-9071-95b5fec8d0f6.png)

Notes:

 - In practice, matching the `SceneInspector` presentation, directly next to an editable widget was confusing, the only discernible difference was the slight emboss on the editable widget.
 - The `SceneView Inspector` has pioneered the 'dark background for non-editable, normal bg for editable' approach. We're going to need more of this as `EditScope` workflows extend throughout Gaffer. We use this here to differentiate the computed variations count from editable widgets.
 - Im not 100% on having the bevel on the frame, as it steers it back towards looking more 'editable' again.
 - We would then migrate the `SceneInspector` value presentation over to this appearance.